sdk/core/Azure.Core/src/Serialization/ModelJsonConverter.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// A generic converter which allows <see cref="JsonSerializer"/> to be able to serialize and deserialize any models that implement <see cref="IModelJsonSerializable{T}"/>.
+    /// </summary>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+#if !NET5_0 // RequiresUnreferencedCode in net5.0 doesn't have AttributeTargets.Class as a target, but it was added in net6.0
+    [RequiresUnreferencedCode("The constructors of the type being deserialized are dynamically accessed and may be trimmed.")]
+#endif
+    public class ModelJsonConverter : JsonConverter<IModelJsonSerializable<object>>
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+    {
+        /// <summary>
+        /// Gets the <see cref="ModelSerializerOptions"/> used to serialize and deserialize models.
+        /// </summary>
+        public ModelSerializerOptions ModelSerializerOptions { get; }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ModelJsonConverter"/> with a default format of <see cref="ModelSerializerFormat.Json"/>.
+        /// </summary>
+        public ModelJsonConverter()
+            : this(ModelSerializerFormat.Json) { }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ModelJsonConverter"/>.
+        /// </summary>
+        /// <param name="format"> The format to serialize to and deserialize from. </param>
+        public ModelJsonConverter(ModelSerializerFormat format)
+            : this(ModelSerializerOptions.GetOptions(format)) { }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ModelJsonConverter"/>.
+        /// </summary>
+        /// <param name="options">The <see cref="ModelSerializerOptions"/> to use.</param>
+        public ModelJsonConverter(ModelSerializerOptions options)
```

Comment:
Do we need both of the format and options constructor in this version?  We can always add one later if needed, and fewer in the initial release reduces complexity for customers.

Reply:
This is for consistency on our public API we added both overloads here and ModelSerializer